### ISSUE TITLE
Add support for translating between character offsets and code point indices

### ISSUE THREAD

**Issue by wooorm** [vfile/vfile-location#14]
### Initial checklist

- [X] I read the [support docs](https://github.com/vfile/.github/blob/main/support.md)
- [X] I read the [contributing guide](https://github.com/vfile/.github/blob/main/contributing.md)
- [X] I agree to follow the [code of conduct](https://github.com/vfile/.github/blob/main/code-of-conduct.md)
- [X] I searched [issues](https://github.com/search?q=user%3Avfile&type=Issues) and couldn’t find anything (or linked relevant results below)

### Problem

Related: https://github.com/syntax-tree/unist/issues/99.

remark-lint where a user provides a max size of say 80, and they don’t mean the way that JS stores strings.
To check if things are (not) allowed at that point, a lint tool needs to be aware of that difference, and be able to translate between the two.

### Solution

Two new methods: `toCodePointIndex(offset: number): number` / `fromCodePointIndex(index: number): number`

### Alternatives

Different utility.

**Comment by remcohaszing** [comment]
> remark-lint where a user provides a max size of say 80

I think you mean a max line length by this? In that case, I think you just need to calculate the code point index on the current line, not of the whole file.

**Comment by wooorm** [comment]
Yes, it would possible in that case to only look at single lines.

This package can be passed single lines; if that remark-lint rule or other potential use-cases want to implement shortcuts like that, they can. The conversion still needs to be implemented somewhere.

**Comment by wooorm** [comment]
For `remark-lint`, which was used as the use case for this feature, this was not needed after all: https://github.com/remarkjs/remark-lint/commit/ab5954d4ca801c87e6273f793cdd62529200dd73.

It might still be an interesting feature, also because of the closed-as-unresolved syntax-tree/unist#99, at some point in the future.
